src/tools/compiletest/src/header.rs

```diff
@@ -948,6 +951,13 @@ impl Config {
                         raw,
                         testfile.display()
                     );
+                } else if FORBIDDEN_REVISION_NAMES.contains(&revision.as_str()) {
```

Comment:
Problem [LIMITED 1/2]: this should limit the rejection scope only for test suites that use `FileCheck`-based annotations. Notably, this includes assembly, codegen and mir-opt. We should not be rejecting this for test suites that don't use `FileCheck`.

I.e. this check needs to be guarded with something like `matches!(self.mode, Mode::Assembly | Mode::Codegen | Mode::MirOpt)`.